From eb7218fc1b993fa527bb9bbde5aeffeefb371d34 Mon Sep 17 00:00:00 2001
From: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
Date: Wed, 6 Dec 2017 18:56:59 +0530
Subject: [PATCH] drm/amdkfd: gfx9 preempt queues after VM_FAULT

Context-save after a VM_FAULT is required for the debugger.

Change-Id: I4a07448143c161f8ab48c41635cceaa4e13efb94
Signed-off-by: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>

Signed-off-by: Pavan Kumar Ramayanam <pavan.ramayanam@amd.com>
---
 drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
index 42b85e6..06328d7 100644
--- a/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
+++ b/drivers/gpu/drm/amd/amdkfd/kfd_device_queue_manager.c
@@ -1687,7 +1687,7 @@ int kfd_process_vm_fault(struct device_queue_manager *dqm,
 		return -EINVAL;
 	pdd = kfd_get_process_device_data(dqm->dev, p);
 	if (pdd)
-		ret = process_evict_queues(dqm, &pdd->qpd, true);
+		ret = process_evict_queues(dqm, &pdd->qpd, reset);
 	kfd_unref_process(p);
 
 	return ret;
-- 
2.7.4

